pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -425,18 +425,19 @@ protected void setServletContext(ServletContext servletContext) {
         this.servletContext = servletContext;
     }
 
+    protected PartitionedTopicMetadata getPartitionedTopicMetadata(TopicName topicName,
+                                                                   boolean authoritative,
+                                                                   boolean checkAllowAutoCreation) {
+        return getPartitionedTopicMetadataAsync(topicName, authoritative, checkAllowAutoCreation).join();
```

Comment:
Maybe we need to use ``get()`` with a timeout to avoid a long time blocking and unchecked exceptions.

Reply:
yes, right.